pymc3/tests/test_distributions_random.py

```diff
@@ -688,3 +689,35 @@ def ref_rand(size, w, mu, sd):
                      'sd': Domain([[1.5, 2., 3.]], edges=(None, None))}, 
                      size=1000,
                      ref_rand=ref_rand)
+=======
+    def test_density_dist(self):
+        def ref_rand(size, mu, sd):
+            return st.norm.rvs(size=size, loc=mu, scale=sd)
+        
+        class TestDensityDist(pm.DensityDist):
+
+            def __init__(self, **kwargs):
+                norm_dist = pm.Normal.dist()
+                super(TestDensityDist, self).__init__(logp=norm_dist.logp, random=norm_dist.random)
+
+        pymc3_random(TestDensityDist, {},ref_rand=ref_rand)
+
+        def check_model_samplability(self):
+            model = pm.Model()
+            with model:
+                normal_dist = pm.Normal.dist()
+                density_dist = pm.DensityDist('density_dist', normal_dist.logp, random=normal_dist.random)
```

Comment:
This looks great. Does it work with a scipy distribution inside a lambda as well?

Reply:
I didn't quite get that, can you elaborate on that a little? 
My interpretation is something like 
```
def mynormal_logp():
	norm_dist_logp = st.norm.logpdf
	norm_dist_random = np.random.normal
	density_dist = pm.DensityDist('density_dist', norm_dist_logp, random=norm_dist_random)
```
is that what you meant?

Reply:
Exactly.

Reply:
I'll test this locally and check! Btw does PyMC3 handle scipy distributions in general?